### DISCUSSION TITLE
TTS read aloud

### DISCUSSION THREAD

**Discussion by Pantyhose-X** [fcitx/fcitx5#439] [Ideas]
fcitx5 add TTS read aloud function
TTS+keyboard
* Applied to language learner. Language worker's
* Applied to language learning. Pronunciation correction，
* Application to learning a foreign language.  Example Russian German I don't know how to speak, TTS taught me to speak
* Text-to-Speech Always speak typed keys speak Phonetic letters are spoken when long pressing on keyboard keys and navigating by character.

example，
https://www.youtube.com/watch?v=YPBn6D6pf78&t=207s

https://user-images.githubusercontent.com/42504385/127757873-c3c076d3-ce73-40b1-8f57-423459fc0c2f.MP4

https://play.google.com/store/apps/details?id=com.google.android.marvin.talkback

https://user-images.githubusercontent.com/42504385/132130706-a1a8ec10-416a-4711-b5d3-c91086e9710f.mp4

# Additional context
[Mozilla's Text-to-Speech](https://github.com/mozilla/TTS)  
[coqui-ai TTS](https://github.com/coqui-ai/TTS) 
[Google Text-to-Speech apk](https://play.google.com/store/apps/details?id=com.google.android.tts&hl=en_US&gl=US)

**Comment by trmdi** [comment]
What about speech to text? Like when you talk in youtube's search box, it will be turned into text. It's super convenient.

Another idea, translate after typing, like Gboard on Android.

**Comment by wengxt** [comment]
I'd say I like this idea, though, on android, the talk back is not implemented via keyboard, but with a standalone service that listening to the text changes.

But if you consider the state of a11y and tons of different toolkit on linux.. it might not be bad idea to implement this via input method.
